subxt/src/events/events_client.rs

```diff
@@ -58,10 +58,10 @@ where
         // return a Future that's Send + 'static, rather than tied to &self.
         let client = self.client.clone();
         async move {
-            // If a block ref isn't provided, we'll get the latest best block to use.
+            // If a block ref isn't provided, we'll get the latest finalized block to use.
             let block_ref = match block_ref {
                 Some(r) => r,
-                None => client.backend().latest_best_block_ref().await?,
+                None => client.backend().latest_finalized_block_ref().await?,
```

Comment:
Yeah exactly; when I went over the code, it felt like using the current best block for this stuff by default was weird for that exact reason that it may well be pruned. So yeah, this will hopefully be a little more reliable, and of course the user can still provide a block hash that they prefer :)